### PR TITLE
[C-416] Fix swiping behavior with tabs/notifications on android

### PR DESCRIPTION
**Pull request by sliptype**
### Description
* Prevents the notification drawer from opening when swiping tabs on android. Luckily android doesn't have the same nondeterminism in gesture handlers so I was able to do this completely in js, didn't have to touch native android code

https://user-images.githubusercontent.com/19916043/167898569-99c9f7c1-4924-42e9-8bcd-0416ab30c861.mp4

* Refactors the NotificationDrawer context to provide the actual drawer navigation as well as the drawer helpers

### Dragons

Toggling the notification drawer on and off has a slight delay so occasionally if swiping very quickly it can still open when it shouldn't. It's pretty minor though

### How Has This Been Tested?

android and ios

### How will this change be monitored?

TestFlight / play store beta

### CODE REVIEW

File: packages/mobile/src/screens/notifications-screen/content/User.tsx [hunk 1]
```diff
@@ -24,8 +24,8 @@ type UserProps = {
 
 const User = ({ user }: UserProps) => {
   const dispatch = useDispatch()
-  const drawerNavigation = useContext(NotificationsDrawerNavigationContext)
-  const navigation = useNavigation({ customNativeNavigation: drawerNavigation })
+  const { drawerHelpers } = useContext(NotificationsDrawerNavigationContext)
+  const navigation = useNavigation({ customNativeNavigation: drawerHelpers })
```
Comment (dylanjeffers) [inline packages/mobile/src/screens/notifications-screen/content/User.tsx:28]: this customNativeNavigation prop is quite cool

**Review by dylanjeffers — APPROVED** [review]
looks great, so sweet figuring this out

**Review by Kyle-Shanks — APPROVED** [review]
🔥

File: packages/mobile/src/screens/app-screen/AppTabScreen.tsx [hunk 2]
```diff
@@ -106,7 +102,12 @@ export const AppTabScreen = ({ baseScreen, Stack }: AppTabScreenProps) => {
             // If coming from notifs allow swipe to open notifs drawer
             drawerNavigation?.setOptions({ swipeEnabled: !!isFromNotifs })
           } else {
-            drawerNavigation?.setOptions({ swipeEnabled: true })
+            // If on the first tab (or the first stack screen isn't a tab navigator),
+            // enable the drawer
+            const isOnFirstTab = !e?.data?.state.routes[0].state?.index
```
[the remark this answers is not in the document]
Comment (sliptype) [reply packages/mobile/src/screens/app-screen/AppTabScreen.tsx:107]: Yup, exactly!

**Pull request merged**